lib/parsers/moov_parser.rb

```diff
@@ -161,5 +165,36 @@ def parse_sample_description_atom(decoder, atom_tree)
     end
   end
 
-  FormatParser.register_parser new, natures: :video, formats: FTYP_MAP.values, priority: 3
+  def cr3(decoder, atom_tree, fallback_width, fallback_height)
```

Comment:
I'm not sure if it's correct to have this in the moov parser as opposed to using a separate parser. To me it seems like the logic is quite separate and it's just reusing the atom decoder (not that I know much aout the file format so maybe cr3 files are truly moov files 🤷).

For example the fallback width and height here seem to be very video specific using decoder.find_video_trak_atom which does not seem like it should be used for an image.

Reply:
CR3 files do adapt the [ISO base media file format](https://en.wikipedia.org/wiki/ISO_base_media_file_format), which is where the `moov` container comes from, and only includes a minor extension on the traditional format to include some Canon-specific metadata - so I thought it was appropriate, but I see the argument for the contrary (especially with your `AtomDecoder` comment in mind).

To your second point, CR3 files _do_ (oddly) have video `trak` atoms, despite them describing single frames of the same image with different properties (dimensions, dual-pixels, compression, etc.). Agree that it's weird, but it does apply and would serve as an appropriate fallback if we can't find the CMT1 atom for some reason to parse the EXIF there.